src/progressbar/progressbar.ts

```diff
@@ -0,0 +1,50 @@
+import {Component, Input, ElementRef, Renderer} from 'angular2/angular2';
+
+@Component({
+  selector: 'ngb-progressbar',
+  template: `
+    <progress class="progress {{class}} {{type && 'progress-' + type}}" [value]="value" [max]="max">
+      <div class="progress">
+        <span class="progress-bar" [style.width.%]="calculatePorcentage()"><ng-content></ng-content></span>
+      </div>
+    </progress>
+  `
+})
+export class NgbProgressbar {
+  _value: number;
+  _class: string;
+  min = 0;
+
+  @Input() set class
+  (newClasses: string) {
+    this._class = newClasses;
+    const initialClasses = newClasses && newClasses.split(' ');
+    initialClasses.forEach(className => this.toggleClass(className));
```

Comment:
Sure, that is for toggling out (remove) the classes in the host element. That is what ng-class does too. The classes are still on host until you remove them, and I remove them in the same way as ng-class, with a toggleClass method that will modify the host.